samples/admin/ADMINReqMoveComplete/ADMINReqMoveComplete.c

```diff
@@ -0,0 +1,500 @@
+        return (1);
+    }
+
+    NSFNoteClose (nhNote);
+    PRINTLOG("!!! ADMINReqMoveUserInHier Processed Sucessfully !!!\n ");
+
+    PRINTLOG("\n*************************************************\n");
+    PRINTLOG("*************Testing-ADMINReqMoveComplete********\n");
+    PRINTLOG("*************************************************\n");
+
+    /* Find the view ID for viewname. */
+    if (error = NIFFindView(
+                   db_handle, 
+                   szViewName, 
+                   &viewID))
+    {
+	NSFDbClose (db_handle);
+	PRINTERROR (error,"NIFFindView");
+	NotesTerm();
+	return (1);
+    }	
+	
+    /* Open the Collection. */
+    if (error = NIFOpenCollection(db_handle,
+					db_handle,
+					viewID,
+					0,
+					NULLHANDLE,
+					&colHandle,
+					NULLHANDLE,
+					NULL,
+					NULLHANDLE,
+					NULLHANDLE))
+    {
+	NSFDbClose (db_handle);
+	PRINTERROR (error,"NIFOpenCollection");
+	NotesTerm();
+	return (1);
+    }
+	
+    colPosition.Level = 0;
+    colPosition.Tumbler[0] = 0;
+
+    /* Get a buffer of all the note IDs for the notes in this collection.
+       Check the return code to see if the collection is empty. */
+
+    do
+    {
+	error = NIFReadEntries(
+			colHandle,       /* handle to this collection */
+			&colPosition,    /* where to start in collection */
+			NAVIGATE_NEXT,   /* order to skip entries */
+			1L,              /* number to skip */
+			NAVIGATE_NEXT,   /* order to use after skipping */
+			0xFFFF,          /* max return number */
+			READ_MASK_NOTEID,/* info we want */
+			&hBuffer,        /* handle to info (return) */
+			NULL,            /* length of buffer (return) */
+			NULL,            /* entries skipped (return) */
+			&dwEntriesFound, /* number of notes (return) */
+			&wSignalFlag) ;  /* share warning (return) */
+
+	if (error) 
+	{
+	    NIFCloseCollection (colHandle);
+	    NSFDbClose (db_handle);
+	    PRINTERROR (error,"NIFReadEntries");
+	    NotesTerm();
+	    return (1);
+	}
+		
+
+	/* Lock down (freeze the location) of the buffer of notes IDs. Cast
+	   the resulting pointer to the type we need. */
+
+	if (hBuffer == NULLHANDLE)
+	{
+	    PRINTLOG ("\nEmpty buffer returned by NIFReadEntries.\n");
+	    NIFCloseCollection (colHandle);
+	    NSFDbClose (db_handle);
+	}
+
+	IdList = (NOTEID far*)OSLockObject(hBuffer);
+
+	if(IdList == NULL)
+	{
+	    PRINTLOG ("\nNote ID list is Empty.\n");
+	    NIFCloseCollection (colHandle);
+	    NSFDbClose (db_handle);
+            OSUnlockObject(hBuffer);
+            OSMemFree(hBuffer);
+            NotesTerm();
+            return (1);
+	}
+	else {
+	/* Print out the list of all the note IDs in this collection. */
+
+	    for (i=0; i<dwEntriesFound; i++)
+	    {
+	         if (NOTEID_CATEGORY & IdList[i])
+			    continue;
+	         PRINTLOG ("Note count is %lu. \t Note ID is: %lX\n",
+                 ++notesFound, IdList[i]);
+	    }
+
+	    NOTEID nid = IdList[1];
+            if (error = NSFNoteOpen(db_handle, nid, 0, &nhNote))
```

Comment:
What is the use of this code NSFNoteOpen for nid which is IdList[1]

Reply:
We just need ONLY one noteID present in the "move name requests" view. I am going to break out once I get the first noteID . Please note that the documents added here will ONLY be by "ADMINReqMovUserInHier" API . So we are sure that we need this noteID.